metadata-ingestion/src/datahub/ingestion/source/state_provider/file_ingestion_checkpointing_provider.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import pathlib
+from datetime import datetime
+from typing import Any, Dict, List, Optional, cast
+
+from datahub.configuration.common import ConfigurationError
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.ingestion_job_checkpointing_provider_base import (
+    IngestionCheckpointingProviderBase,
+    IngestionCheckpointingProviderConfig,
+    JobId,
+)
+from datahub.ingestion.sink.file import write_metadata_file
+from datahub.ingestion.source.file import read_metadata_file
+from datahub.metadata.schema_classes import DatahubIngestionCheckpointClass
+
+logger = logging.getLogger(__name__)
+
+
+class FileIngestionStateProviderConfig(IngestionCheckpointingProviderConfig):
+    filename: Optional[str] = None
+
+
+class FileIngestionCheckpointingProvider(IngestionCheckpointingProviderBase):
+    orchestrator_name: str = "file"
+
+    def __init__(self, filename: str, name: str):
+        super().__init__(name)
```

Comment:
instead of taking name as an arg, use `name=self.__class__.__name__`

feel free to change `DatahubIngestionCheckpointingProvider`, and remove the `name` parameter from the `IngestionCheckpointingProviderBase.__init__` and create methods